gemini/sample-apps/fashion-trends-prediction/app/data_prep/driver.py

```diff
@@ -0,0 +1,331 @@
+import json
+import time
+
+from config import config
+from dataProcessing import getPosts
+from helper_functions_insta import getInfluencers, instalogin
+from helper_functions_vogue import get_articles
+from prepare_data_for_retriever import prepare_data_for_retriever
+
+parameters = config["parameters"]["standard"]
+country_names = config["countryList"]
+data_path = config["Data"]["current_data"]
+
+
+def isRoot(word):
+    if word.strip() == word and word.lower() == word:
+        return True
+    return False
+
+
+def cmpf(item):
+    return -len(item[1])
+
+
+def extract_json_influencer_wise(scrape_size, first_scrape, saved):
+    """Extracts data from Instagram and saves it in a JSON file.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+            first_scrape (bool): A boolean indicating whether this is the first scrape.
+            saved (dict): A dictionary containing the saved data.
+
+    """
+
+    if first_scrape:
+        temp_map = {}
+        scraped_influencers = set()
+    else:
+        with open(data_path, "r") as f:
+            temp_map = json.load(f)
+        scraped_influencers = set(temp_map.keys())
+
+    cookies = instalogin()
+
+    for country_name in country_names[: scrape_size["num_countries"]]:
+        print(country_name)
```

Comment:
Are most of these print statements necessary?

Reply:
It takes a lot of time to scrape so the print statements help to keep track. Can be removed if necessary.

Reply:
I would recommend using logging statements instead so they can be hidden more easily.

Reply:
Added logging statements in the data_prep files